vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
+    "bar": { "baseline": "2024-08-01", "port-version": 1 },
+    "baz": { "baseline": "vista-xp", "port-version": 0 },
+  }
+}
+```
+
+See the reference documentation for:
+
+* baseline file layout for [Git
+  registries](../maintainers/registries.md#git-baseline)
+* baseline file layout for [filesystem
+  registries](../maintainers/registries.md#filesystem-baseline)
+
+## Version files
+
+Each port in the registry must have a corresponding versions file. The versions
+file is a JSON file named the same as its corresponding port, for example, a
+port named `foo` must have a corresponding `foo.json` file. 
+
+vcpkg expects the versions files to be stored in the following location
+`versions/<prefix>/<port name>.json` , where `<prefix>` is the first letter of
+the port name followed by a hyphen. For example, the versions file for port
+`foo` must be stored in `versions/f-/foo.json`.
+
+The purpose of the versions file is two-fold:
+
+* List all available versions of each port
+* Point to the retrieval locations of each of these versions.
+
+The layout of the version file is an object containing a `"versions"` array, with
+each entry in that array being a version object.  A version object must contain
+the following properties: 
+
+* A version property: The property's key and value must match the ones used by
+  the port in its `vcpkg.json` file. The key must be one of `version`,
+  `version-semver`, `version-date`, or `version-string`; the value must be the
+  version as it appears in the port's manifest file (`vcpkg.json`). 
+* `port-version`: the value is the port's `port-version` as it appears in the
+  port's `vcpkg.json` file. 
+* `git-tree`: (only on Git registries) the value is the git-tree SHA
+  corresponding to the port's directory. This is a SHA calculated by hashing the
+  contents of the port's directory; this git-tree SHA can be used by Git to
+  retrieve the contents of the port matching the provided git-tree. This makes
+  it possible for vcpkg to retrieve old versions of ports from the registries
+  Git history. Read the [git registries](#git-registries) section to learn how
+  to obtain this value for any given port version.
+* `path`: (only on filesystem registries) the value is the full path to a
+  directory containing the port files for the specific version.
+
+## Example of a filesystem registry version file
+
+```json
+{
+  "versions": [
+    {
+      "path": "$/ports/foo/1.2.0",
+      "version": "1.2.0",
+      "port-version": 0
+    },
+    {
+      "path": "$/ports/foo/1.1.0",
+      "version": "1.1.0",
+      "port-version": 0
+    },
+    {
+      "path": "$/ports/foo/1.0.0",
+      "version": "1.0.0",
+      "port-version": 0
+    }
+  ]
+}
+``` 
+
+See the reference documentation for:
+
+* version file layout for [Git
+  registries](../maintainers/registries.md#git-version-file)
+* version file layout for [filesystem
+  registries](../maintainers/registries.md#filesystem-version-file)
 
 ## Built-in registry
```

Comment:
Given that the user can't name this thing, do we want to describe it as a registry in the first place? It breaks a lot of the rules you've described elsewhere in this document...

Reply:
I think this is important to describe what happens in classic mode and when users don't explicitly set a default registry.